pkg/tbtc/tbtc.go

```diff
@@ -0,0 +1,62 @@
+package tbtc
+
+import (
+	"context"
+	"fmt"
+	"github.com/ipfs/go-log"
+	"github.com/keep-network/keep-common/pkg/persistence"
+	"github.com/keep-network/keep-core/pkg/net"
+	"github.com/keep-network/keep-core/pkg/sortition"
+)
+
+var logger = log.Logger("keep-tbtc")
+
+// Initialize kicks off the TBTC by initializing internal state, ensuring
+// preconditions like staking are met, and then kicking off the internal TBTC
+// implementation. Returns an error if this failed.
+func Initialize(
+	ctx context.Context,
+	chain Chain,
```

Comment:
I would consider renaming it: `chain` -> `tbtcChain`, so that we know quickly we are dealing with a tbtc chain here.
Also in the `Initialize` function in `beacon` package the interface is named differently:
```
func Initialize(
	ctx context.Context,
	beaconChain beaconchain.Interface,
	netProvider net.Provider,
	persistence persistence.Handle,
)
```

We could settle on one name in both beacon and tbtc chains (`Chain` vs `Interface`).

Reply:
Regarding the `tbtcChain` rename, I think that's redundant because we are dealing only with the TBTC chain. Nothing else.

Regarding `beaconchain.Interface` from `pkg/beacon`, the story is not that simple. That package relies on a legacy structure we used before and if we want to have `beacon.Chain` instead of `beaconchain.Interface` we would have to move `chain.go` from `pkg/beacon/chain` to the root `pkg/beacon`. However, that causes a lot of import cycles that needs to be solved during a bigger refactoring. We are considering such a refactor but there are items with higher priority now.